logproof/src/linear_algebra.rs

```diff
@@ -12,13 +12,28 @@ use crate::fields::{FieldFrom, FpRistretto};
 use crate::math::{ModSwitch, One, Rem, SmartMul, Tensor, Zero};
 
 #[derive(Debug, Clone, PartialEq)]
+/**
+ * An `m x n` matrix of elements.
+ *
+ * # Remarks
+ * Matrix elements can be any type that implements [`Add`], [`Mul`],
+ * [`Clone`], [`Zero`].
+ */
 pub struct Matrix<T>
 where
     T: Zero + Clone,
```

Comment:
I'm guessing there is no way to add the add and multiply constraint easily?

Reply:
Added clarity. Add and SmartMul aren't strictly required, but they enable more operations.